File: Factory Pattern/Cpp/Pizza/NYStylePizza/NYStyleVeggiePizza.cpp
#include "../Pizza.cpp"

class NYStyleVeggiePizza : public Pizza {
public:
    NYStyleVeggiePizza() {
        name = "뉴욕 스타일 소스와 채소 피자";
        dough = "씬 크러스트 도우";
        sauce = "마리나라 소스";

        toppings.push_back("\t잘게 썬 채소\n");
    }
};

File: Patterns/Factory Method Pattern/CSharp/Pizza/NYStylePizza/NYStyleVeggiePizza.cs
using System;

namespace Pizza.NYStylePizza
{
    public class NYStyleVeggiePizza : Pizza 
    {
        public NYStyleVeggiePizza() :
            base("뉴욕 스타일 소스와 채소 피자", "씬 크러스트 도우", "마리나라 소스")
        {
            toppings.Add("\t잘게 썬 채소");
        }
    }
}